File: src/test/java/com/codeborne/json/JsonAssertTest.java
package com.codeborne.json;

import org.junit.jupiter.api.Test;

import java.util.Map;

import static com.codeborne.json.JsonAssertions.assertThat;

class JsonAssertTest {
  @Test
  void canExtractElementsFromMap() {
    assertThat(Map.of("user", Map.of("name", "me")))
      .extractingFromJson("user", "name").isEqualTo("me");
  }
}

File: src/test/java/com/codeborne/json/assertions/JsonAssertTest.kt
package com.codeborne.json.assertions

import org.junit.jupiter.api.Test
import java.util.Map

internal class JsonAssertTest {
    @Test
    fun canExtractElementsFromMap() {
        JsonAssertions.assertThat(Map.of("user", Map.of("name", "me")))
            .extractingFromJson("user", "name").isEqualTo("me")
    }
}
